metrics/stackdriver_exporter.go

```diff
@@ -63,10 +75,31 @@ func newOpencensusSDExporter(o stackdriver.Options) (view.Exporter, error) {
 // TODO should be properly refactored to be able to inject the getMonitoredResourceFunc function.
 // 	See https://github.com/knative/pkg/issues/608
 func newStackdriverExporter(config *metricsConfig, logger *zap.SugaredLogger) (view.Exporter, error) {
-	gm := gcpMetadataFunc()
+	gm := getGcpMetadata(config)
 	mtf := getMetricTypeFunc(config.stackdriverMetricTypePrefix, config.stackdriverCustomMetricTypePrefix)
+
+	proj := config.stackdriverProjectID
+	if config.stackdriverConfig.ProjectID != "" {
+		proj = config.stackdriverConfig.ProjectID
+	}
+
+	var co []option.ClientOption
+	if config.stackdriverConfig.GcpSecretName != "" {
+		secret, err := getStackdriverSecretFunc(&config.stackdriverConfig)
+
+		if err == nil {
+			co = []option.ClientOption{sdconfig.ConvertSecretToExporterOption(secret)}
+		} else {
+			logger.Errorf("Failed to retrieve Secret for authenticating with Stackdriver: ", zap.Error(err))
+			// Automatically fall back on Google application default credentials
+		}
+	}
+
 	e, err := newStackdriverExporterFunc(stackdriver.Options{
-		ProjectID:               config.stackdriverProjectID,
+		ProjectID:               proj,
+		Location:                config.stackdriverConfig.GcpLocation,
```

Comment:
Could you please add a comment that if `config.stackdriverConfig.GcpLocation` is empty, automatically fall back on the location provided by the metadata server?

Reply:
Pending